gensim/models/experimental/drmm_tks.py

```diff
@@ -0,0 +1,595 @@
+            The method for handling unkown words
+            'zero': unknown words are given a zero vector
+            'random': unknown words are given a uniformly random vector
+
+        validation_data: list of the form [test_queries, test_docs, test_labels]
+            where test_queries, test_docs  and test_labels are of the same form as
+            their counter parts stated above
+
+        """
+        self.queries = queries
+        self.docs = docs
+        self.labels = labels
+        self.word_counter = Counter()
+        self.text_maxlen = text_maxlen
+        # self.hist_size = hist_size
+        self.word_embedding_path = word_embedding_path
+        self.word2index, self.index2word = {}, {}
+        self.keep_full_embedding = keep_full_embedding
+        self.normalize_embeddings = normalize_embeddings
+        self.model = None
+        self.epochs = epochs
+        self.validation_data = validation_data
+
+        if unk_handle_method not in ['random', 'zero']:
+            raise ValueError("Unkown token handling method %s" %
+                             str(unk_handle_method))
+        self.unk_handle_method = unk_handle_method
+
+        self.build_vocab()
+        self.pair_list = self.get_pair_list()
+        self.indexed_pair_list = self.make_indexed_pair_list()
+        self.train_model()
+
+    def build_vocab(self):
+        """Indexes all the words and makes an embedding_matrix which
+        can be fed directly into an Embedding layer"""
+
+        logger.info("Starting Vocab Build")
+
+        # get all the vocab words
+        for q in self.queries:
+            self.word_counter.update(q)
+        for doc in self.docs:
+            for d in doc:
+                self.word_counter.update(d)
+        for i, word in enumerate(self.word_counter.keys()):
+            self.word2index[word] = i
+            self.index2word[i] = word
+
+        self.vocab_size = len(self.word2index)
+        logger.info("Vocab Build Complete")
+        logger.info("Vocab Size is %d" % self.vocab_size)
+
+        logger.info("Building embedding index using pretrained word embeddings")
+        # Use KeyedVectors for easy and quick access od word embeddings
+        glove_file = self.word_embedding_path
+        tmp_file = get_tmpfile("tmp_word2vec.txt")
+        embedding_vocab_size, self.embedding_dim = glove2word2vec(
+            glove_file, tmp_file)
+        kv_model = KeyedVectors.load_word2vec_format(tmp_file)
+
+        logger.info("The embeddings_index built from the given file has %d words of %d dimensions" %
+                    (embedding_vocab_size, self.embedding_dim))
+
+        logger.info(
+            "Building the Embedding Matrix for the model's Embedding Layer")
+
+        # Initialize the embedding matrix
+        # UNK word gets the vector based on the method
+        if self.unk_handle_method == 'random':
+            self.embedding_matrix = np.random.uniform(-0.2, 0.2,
```

Comment:
make an numpy random generator and use it

Reply:
also I still see no seeding by each word

Reply:
and still no seeding by each word

Reply:
> and still no seeding by each word

It is there

https://github.com/aneesh-joshi/gensim/blob/2ac7f3153e46ee1281b67392bcb8f66f93708a01/gensim/models/experimental/drmm_tks.py#L271